projects/WTI-UI/src/app/modules/runs/components/new-run/new-run.component.ts

```diff
@@ -196,6 +199,34 @@ export class NewRunComponent implements OnInit, OnDestroy {
 	//none of the specified FileSubmission files contains a space in its name
 	return false;
 	}
+
+  //returns true if there are duplicate filenames among any of the specified FileSubmissions; false if all unique.
+  private filenameContainsDuplicates(mainfile: FileSubmission, additionalFiles: FileSubmission []): boolean {
+    let mainfilename = mainfile.fileName;
+    let l = mainfilename.indexOf("/");
+    if (l == -1) l = mainfilename.indexOf("\\");
+    if (l > -1) mainfilename = mainfilename.substring(l + 1);
```

Comment:
I do not understand the testing for `/` and `/` characters in the file name.  My quick testing (I printed out the filenames on the browser console inside of this method) indicates that what is being passed into the component (and hence into this method) is the `basename` of the file (i.e., the file name with all paths elements removed).  If this is the case, there's no point in looking for a `/` or `\`.

On the other hand, if there WAS a `/` or `\` in the file name, it's quite likely there would be MULTIPLE such characters (e.g. /usr/me/mydir/filename).  In that case, `indexOf("/")` would return the index of the FIRST such character -- and the following `substring(l+1)` would return a string which still has all the remaining `/` (or `\`) characters in it -- which wouldn't end up comparing the basename.     What's the purpose of doing it this way?

(Same comments apply to the testing of the "additional file" names, below...)

Have I somehow misunderstood what this code is doing?

Reply:
You're correct it is a bug. I had meant to use the function `lastIndexOf()` instead of `indexOf()`. However, when I tested it didn't have erroneous behavior possibly because as you mentioned it is only taking the `basename`.

So, I wanted to know will it always be the case that only `basename` of the file taken. If yes, then I can get rid of all the checks for file separator character.  Otherwise, I can change it to `lastIndexOf()`. Also if I have to use the `path.sep` it needs to be further explored how to solve the `error TS2307: Cannot find module 'path' or its corresponding type declarations` while building the `package.xml`

Reply:
We just got finished discussing this very issue on the PC2 call, and I think we decided that you will only get the main file basename, and the list of additional files will only be basenames as well, however, we'd like to leave the code in there that deals with a path name if there are separators.

Reply:
What I would suggest regarding the `path.sep` issue is to file a separate GitHub issue pointing out that it's a fix that should (eventually) be applied to `new-run.component.ts`, and don't worry about it any further for now.  I think it's likely to be a rabbit-hole that will just distract people (Samanway) from moving on to other things which are more important at the moment.